deep_qa/data/instances/mc_tuples_with_background_instance.py

```diff
@@ -0,0 +1,358 @@
+    # Return:
+    #   indexed: List[List[List[int]]]
+    def index_tuples(self, tuples_in: List[TextTuple], data_indexer: DataIndexer):
+        indexed = []
+        for text_tuple in tuples_in:
+            text_list = text_tuple.to_text_list(self.object_handling, self.use_context)
+            # indices: List[List[int]]
+            indices = [self._index_text(phrase, data_indexer) for phrase in text_list]
+            indexed.append(indices)
+        return indexed
+
+    @classmethod
+    def read_from_line(cls, line: str, default_label: int=None):
+        """
+        Reads a TupleInstances from a line.  The format has one option (currently):
+            [question index][tab][num answer coptions][tab][all question+answer tuples][tab]
+            [background tuples]
+        The question+answer tuples are formatted as:
+            [a_1-tuple_1]$$$...$$$[a_1-tuple_n]###...###[a+_m-tuple_1]$$$...$$$||[a_m-tuple_p]
+        such that ``$$$`` serves as the tuple delimiter and ``###`` serves as the answer candidate
+        delimiter.
+        Each of the tuples is formatted as:
+            [s words]|[v words]|[o_1 words]|...|[o_j words]|[context: context words]
+        such that ``|`` serves as the tuple-slot delimiter.  Note that some slots may contain prefixes
+        indicating the type of information they contain (e.g., "context:", see below for more).
+
+        Objects are optional, and can vary in number. This makes the acceptable number of slots per
+        tuple 2 or more.
+
+        If a tuple slot is reserved for context, it contains "context:" at the start.
+
+        Some tuples contain coreference resolution info (from one of the table sources).  We do not
+        currently handle this, but plan to eventually replace the referring expression with its resolved
+        value.
+            Example: it[sky]|is|blue --> would become --> sky|is|blue
+
+        Some tuples additionally contain prefixed information about some of the "objects" such as "L:" for
+        locations and T: for time.  At present this information is simply filtered out, but can be used
+        later.
+        """
+        fields = line.split("\t")
+        assert len(fields) == 5, \
+            "Input line is not the correct number of fields (should be 5, it is {0}) " \
+            "-- line:{1}".format(len(fields), line)
+        # Question index
+        index = int(fields[0])
+        # Number of answer candidates
+        num_answers = int(fields[1])
+        # Answer Candidate tuples
+        answers = fields[2].split("###")
+        answer_tuples = []
+        for curr_answer in answers:
+            curr_answer_tuple_strings = curr_answer.split("$$$")
+            curr_answer_tuples = [TextTuple.from_string(tuple_string)
+                                  for tuple_string in curr_answer_tuple_strings if len(tuple_string) > 0]
+            answer_tuples.append(curr_answer_tuples)
+        assert len(answer_tuples) == num_answers, \
```

Comment:
I just wanted it as a check, since the file is essentially unreadable to a human, in case somewhere between tushar passing me data, me splicing in the answer label from another source, or me loading in the data it got screwed up.